tests/Benchmarks/AzureBench.cs

```diff
@@ -88,5 +112,135 @@ static string CosmosDbBaselineFromMsdn(string key, string keyType, string verb,
 
         return result;
     }
+
+    public static bool TryWritePrimitives(Span<byte> output, Sha256 hash, string keyType, string verb, string resourceId, string resourceType, string tokenVersion, DateTime utc, out int bytesWritten)
+    {
+        int written, consumed, totalWritten = 0;
+        bytesWritten = 0;
+
+        Span<byte> buffer = stackalloc byte[AuthenticationHeaderBufferSize];
+
+        s_type.CopyTo(buffer);
+        totalWritten += s_type.Length;
+        var bufferSlice = buffer.Slice(totalWritten);
```

Comment:
Do we need bufferSlice?
It looks like you always just update totalWritten and do `bufferSlice = buffer.Slice(totalWritten);`
Why not just pass `buffer.Slice(totalWritten)` everywhere instead of setting it to bufferSlice?

Reply:
I can remove some of these. Later in the code, the sliced buffer is reused.